app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -181,55 +187,122 @@ private void requestNotificationPermission() {
     @Override
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
-        if (intent.getExtras() != null){
-            if (intent.getExtras().containsKey(RES_ITEM))
-                binding.bottomNavigation.setSelectedItemId(intent.getIntExtra(RES_ITEM, R.id.dashboard));
-            else if (intent.getExtras().containsKey(NOTIFICATION_DIALOG)){
-                new ConfirmDialogFragment.Builder()
-                        .withTitle(intent.getExtras().getString("title"))
-                        .withMessage(intent.getExtras().getString("message"))
-                        .withNegativeButton("")
-                        .withPositiveButton(getString(R.string.Modal_OK))
-                        .build().show(getSupportFragmentManager(), null);
+        if (Intent.ACTION_VIEW.equals(intent.getAction())) {
+            Uri uri = intent.getData();
+            if (uri == null) return;
```

Comment:
Would you mind not using this style where we collapse if and clause in a single line? I tend to prefer expanding it into N lines with `{` and `}` because it's more readable. (Kotlin has better facilities for being compact and they read easy, while I'd rather stick on scholastic Java for Java code, which risks being more tricky.)